services/core-api/app/__init__.py

```diff
@@ -97,11 +99,94 @@ def register_routes(app):
     root_api_namespace.add_namespace(orgbook_api)
     root_api_namespace.add_namespace(EMLI_contacts_api)
 
-    # Healthcheck endpoint
+    # General Service status
     @root_api_namespace.route('/health')
     class Healthcheck(Resource):
+        def get_health(self):
+            service = {
+                'database': False,
+                'cache' : False,
+                'nris' : False,
+                'docgen': False,
+                'docman': False
+            }
+            status = 200
+
+            try:
+                service['database'] = get_database_status()
+            except Exception as error:
+                app.logger.error("Database health check failed " + str(error))
+                service['database'] = False
+                status = 503
+
+            try:
+                service['cache'] = get_cache_status()
+            except Exception as error:
+                app.logger.error("Cache health check failed " + str(error))
+                service['cache'] = False
+                status = 503
+
+            try:
+                service['nris'] = get_service_status('NRIS_API_URL')
+            except Exception as error:
+                app.logger.error("NRIS health check failed " + str(error))
+                service['nris'] = False
+                status = 503
+
+            try:
+                service['docgen'] = get_service_status('DOCUMENT_GENERATOR_URL')
+            except Exception as error:
+                app.logger.error("Docgen health check failed " + str(error))
+                service['docgen'] = False
+                status = 503
+
+            try:
+                service['docman'] = get_service_status('DOCUMENT_MANAGER_URL')
+            except Exception as error:
+                app.logger.error("Document Manager health check failed " + str(error))
+                service['docman'] = False
+                status = 503
```

Comment:
Having a 503 error if any of the dependencies are not available will be useful for alerting by external monitoring services.